src/java.base/share/classes/sun/security/x509/X509CRLImpl.java

```diff
@@ -280,14 +280,20 @@ public X500Principal getIssuerX500Principal() {
          *   prevCertIssuer if it does not exist
          */
         private X500Principal getCertIssuer(X509CRLEntryImpl entry,
-                                            X500Principal prevCertIssuer) {
+                X500Principal prevCertIssuer) throws CRLException {
 
             CertificateIssuerExtension ciExt =
                     entry.getCertificateIssuerExtension();
             if (ciExt != null) {
                 GeneralNames names = ciExt.getNames();
-                X500Name issuerDN = (X500Name) names.get(0).getName();
-                return issuerDN.asX500Principal();
+                Iterator<GeneralName> itr = names.iterator();
+                while (itr.hasNext()) {
+                    if (itr.next().getName() instanceof X500Name issuerDN) {
+                        return issuerDN.asX500Principal();
+                    }
+                }
+                throw new CRLException("Parsing error: "
+                        + "issuer is not an X.500 DN");
```

Comment:
To be more precise, I suggest changing this message to "CertificateIssuer field does not contain an X.500 DN".